src/app/modules/item/components/item-remove-button/item-remove-button.component.ts

```diff
@@ -0,0 +1,96 @@
+import { Component, Input, OnChanges } from '@angular/core';
+import { ReplaySubject } from 'rxjs';
+import { distinct, map, switchMap } from 'rxjs/operators';
+import { mapToFetchState } from '../../../../shared/operators/state';
+import { GetItemChildrenService, ItemChild } from '../../http-services/get-item-children.service';
+import { Item } from '../../http-services/get-item-by-id.service';
+import { ConfirmationService } from 'primeng/api';
+import { RemoveItemService } from '../../http-services/remove-item.service';
+import { ActionFeedbackService } from '../../../../shared/services/action-feedback.service';
+import { Router } from '@angular/router';
+
+@Component({
+  selector: 'alg-item-remove-button',
+  templateUrl: './item-remove-button.component.html',
+  styleUrls: [ './item-remove-button.component.scss' ],
+})
+export class ItemRemoveButtonComponent implements OnChanges {
+  @Input() item?: Item;
+  @Input() attemptId?: string;
+
+  private readonly params$ = new ReplaySubject<{ id: string, attemptId: string }>(1);
+  readonly state$ = this.params$.pipe(
+    distinct(),
```

Comment:
Does it have any effect :thinking: ?
If `params$` is a subject of an object, every value value is an object, so they are all different no ?

I wonder if it's not a `distinctUntilChanged((a, b) => a.id === b.id && a.attemptId === b.attemptId)` you're looking for.

If it makes to change, go for it :pray: